milli/tests/search/query_criteria.rs

```diff
@@ -0,0 +1,531 @@
+use milli::{Search, SearchResult, Criterion};
+use big_s::S;
+
+use crate::search::{self, EXTERNAL_DOCUMENTS_IDS};
+
+#[test]
+fn none() {
+    let criteria = vec![];
+    let index = search::setup_search_index_with_criteria(&criteria);
+    let mut rtxn = index.read_txn().unwrap();
+
+    let mut search = Search::new(&mut rtxn, &index);
+    search.query(search::TEST_QUERY);
+    search.limit(EXTERNAL_DOCUMENTS_IDS.len());
+
+    let SearchResult { matching_words: _matching_words, candidates: _candidates, documents_ids } = search.execute().unwrap();
+
+    let expected_external_ids: Vec<_> = search::expected_order(&criteria, true, true).into_iter().map(|d| d.id).collect();
```

Comment:
You could maybe introduce two constants to help read the code and use this in all of the next functions.

```suggestion
    const AUTHORIZED_TYPO: bool = true;
    const OPTIONAL_WORDS: bool = true;

    let expected_external_ids: Vec<_> = search::expected_order(
        &criteria,
        AUTHORIZED_TYPO,
        OPTIONAL_WORDS,
    ).into_iter().map(|d| d.id).collect();
```